tradingstrategy/binance/candlestick.py

```diff
@@ -0,0 +1,221 @@
+    open_prices, high_prices, low_prices, close_prices, volume, dates = (
+        [],
+        [],
+        [],
+        [],
+        [],
+        [],
+    )
+
+    for i in range(0, len(timestamps) - 1):
+        start_timestamp = timestamps[i]
+        end_timestamp = timestamps[i + 1]
+        full_params_str = (
+            f"{params_str}&startTime={start_timestamp}&endTime={end_timestamp}"
+        )
+        url = f"https://api.binance.com/api/v3/klines?{full_params_str}&limit=1000"
+        response = requests.get(url)
+        if response.status_code == 200:
+            json_data = response.json()
+            if len(json_data) > 0:
+                for item in json_data:
+                    dates.append(datetime.datetime.fromtimestamp(item[0] / 1000))
+                    open_prices.append(float(item[1]))
+                    high_prices.append(float(item[2]))
+                    low_prices.append(float(item[3]))
+                    close_prices.append(float(item[4]))
+                    volume.append(float(item[5]))
+        else:
+            logger.warn(f"Error fetching data between {start_timestamp} and {end_timestamp}. \nResponse: {response.status_code} {response.text} \nMake sure you are using valid pair symbol e.g. `ETHUSDC`, not just ETH")
+
+    df = pd.DataFrame(
+        {
+            "open": open_prices,
+            "high": high_prices,
+            "low": low_prices,
+            "close": close_prices,
+            "volume": volume,
+        },
+        index=dates,
+    )
+
+    # remove rows with index (date) duplicates
+    df = df[df.index.duplicated(keep="first") == False]
+
+    # df = clean_time_series_data(df)  
+    path = get_parquet_path(symbol, time_bucket, start_at, end_at)
+    df.to_parquet(path)
+
+    return df
+
+
+def get_binance_data_parquet(
+    symbol: str,
+    time_bucket: TimeBucket,
+    start_at: datetime.datetime,
+    end_at: datetime.datetime,
+) -> pd.DataFrame:
+    """Get parquet file for the candlestick data.
+
+    :param symbol: Trading pair symbol E.g. ETHUSDC
+    :param time_bucket: TimeBucket instance
+    :param start_at: Start date of the data
+    :param end_at: End date of the data
+    :return: Path to the parquet file
+    """
+    try:
+        return pd.read_parquet(
+            get_parquet_path(symbol, time_bucket, start_at, end_at)
+        )
+    except:
+        raise ValueError(
+            "Parquet file not found. Use get_binance_candlestick_data to create it."
+        )
+
+
+def get_parquet_path(
+    symbol: str,
+    time_bucket: TimeBucket,
+    start_at: datetime.datetime,
+    end_at: datetime.datetime,
+) -> Path:
+    """Get parquet path for the candlestick data.
+    
+    :param symbol: Trading pair symbol E.g. ETHUSDC
+    :param time_bucket: TimeBucket instance
+    :param start_at: Start date of the data
+    :param end_at: End date of the data
+    :return: Path to the parquet file
+    """
+    return Path(
```

Comment:
This path must be configurable so that for unit testing we can configure the tests to use `/tmp` path.  Also it is important to be able to configure paths in Docker as file system may be read only and need special paths.

A possible solution is to create class `BinanceDownloader` which takes `path` as the `__init__` argument